src/KDSoapClient/KDSoapMessageAddressingProperties.cpp

```diff
@@ -296,7 +320,19 @@ void KDSoapMessageAddressingProperties::writeMessageAddressingProperties(KDSoapN
     Q_UNUSED(messageNamespace);
     Q_UNUSED(forceQualified);
 
-    if (d->destination == predefinedAddressToString(None, d->addressingNamespace)) {
+    bool supportsNoneAddressing = false;
+    switch (d->addressingNamespace) {
+    case Addressing200303:
+    case Addressing200403:
+    case Addressing200408:
+        supportsNoneAddressing = false;
+        break;
+    case Addressing200508:
+        supportsNoneAddressing = true;
+        break;
+    }
```

Comment:
Would it make sense to just check for Addressing200508 in a if statement?

Reply:
I dislike if statements for enums with a fierce passion. More so in this case because it is in fact >= 200508, or at the very least it needs looking at for possible future versions so the compiler complaining about the incomplete switch-case is much desired here.